java/AxivityReader.java

```diff
@@ -220,8 +226,16 @@ private static void readCwaBuffer(ByteBuffer buf, LocalDateTime SESSION_START,
                 }
 
                 // determine time for indexed sample within block
-                LocalDateTime blockTime = getCwaTimestamp((int) blockTimestamp, 
+                LocalDateTime blockTime = getCwaTimestamp((int) blockTimestamp,
                                                     fractional, timeZoneOffset);
+
+                // if SESSION_START not set yet, this is the first block
```

Comment:
As comment says, if `SESSION_START` is not set yet, then this is the first block in the file read so it contains the start time.